app/components/core/add-frame.tsx

```diff
@@ -9,36 +9,18 @@ import {
   PopoverTrigger,
 } from "~/components/ui/popover"
 
-export interface FrameData {
-  id?: string;
-  url: string;
-  width: number;
-  height: number;
-  x: number;
-  y: number;
-}
+export default function AddFrame(props: { userId: string }) {
+  const [previewUrl, setPreviewUrl] = useState("");
 
-interface AddFrameProps {
-  onSave: (frame: FrameData) => Promise<void>;
-}
-
-export default function AddFrame({ onSave }: AddFrameProps) {
-  const [frame, setFrame] = useState<FrameData>({url: "", width: 1, height: 1, x: 0, y: 0});
-  const [showIframe, setShowIframe] = useState(false);
-
-  const handlePreview = (event: any) => {
-    event.preventDefault();
-    setShowIframe(true);
-  };
-
-  // This needs to layout the frames in a grid, which is 12 columns on xl and 4 on md, and 2 on sm, and 1 on xs
   return (
     <Popover>
       <PopoverTrigger asChild>
         <Button variant="outline">Add new frame</Button>
       </PopoverTrigger>
       <PopoverContent className="w-[48rem]">
-        <Form onSubmit={handlePreview}>
+        <Form method="post">
```

Comment:
this will just submit to any action its included on, prob fine most of time. could override that if needed.